src/main/java/com/angelbroking/smartapi/SmartConnect.java

```diff
@@ -1,727 +1,781 @@
+    public User getProfile() {
+        try {
+            String url = routes.get(Constants.SMART_CONNECT_API_USER_PROFILE);
+            return ResponseParser.parseResponse(smartAPIRequestHandler.getRequest(this.apiKey, url, accessToken));
+
+        } catch (SmartAPIException ex) {
+            logger.error(Constants.SMART_API_EXCEPTION_OCCURRED + ex.getMessage(), ex);
+            return null;
+        } catch (IOException ex) {
+            logger.error(Constants.IO_EXCEPTION_OCCURRED + ex.getMessage(), ex);
+            return null;
+        } catch (JSONException ex) {
+            logger.error(Constants.JSON_EXCEPTION_OCCURRED + ex.getMessage(), ex);
+            return null;
+        }
+    }
+
+    /**
+     * Places an order.
+     *
+     * @param orderParams is Order params.
+     * @param variety     variety="regular". Order variety can be bo, co, amo,
+     *                    regular.
+     * @return Order contains only orderId.
+     */
+    public Order placeOrder(OrderParams orderParams, String variety) {
+        try {
+            Validators validator = new Validators();
+            validator.orderValidator(orderParams);
+            String url = routes.get(Constants.SMART_CONNECT_API_ORDER_PLACE);
+            JSONObject params = new JSONObject();
+            params.put(Constants.EXCHANGE, orderParams.getExchange());
+            params.put(Constants.TRADING_SYMBOL, orderParams.getTradingSymbol());
+            params.put(Constants.TRANSACTION_TYPE, orderParams.getTransactionType());
+            params.put(Constants.QUANTITY, orderParams.getQuantity());
+            params.put(Constants.PRICE, orderParams.getPrice());
+            params.put(Constants.PRODUCT_TYPE, orderParams.getProductType());
+            params.put(Constants.ORDER_TYPE, orderParams.getOrderType());
+            params.put(Constants.DURATION, orderParams.getDuration());
+            params.put(Constants.SYMBOL_TOKEN, orderParams.getSymbolToken());
+            params.put(Constants.SQUARE_OFF, orderParams.getSquareOff());
+            params.put(Constants.STOP_LOSS, orderParams.getStopLoss());
+            params.put(Constants.VARIETY, variety);
+            JSONObject jsonObject = smartAPIRequestHandler.postRequest(this.apiKey, url, params, accessToken);
+            if (logger.isDebugEnabled()) {
+                logger.debug(String.valueOf(jsonObject));
+            }
+            Order order = new Order();
+            order.setOrderId(jsonObject.getJSONObject(Constants.SMART_CONNECT_DATA).getString(orderID));
+            if (logger.isDebugEnabled()) {
+                logger.debug(String.valueOf(order));
+            }
+            return order;
+        } catch (SmartAPIException ex) {
+            logger.error(Constants.SMART_API_EXCEPTION_OCCURRED + ex.getMessage(), ex);
+            return null;
+        } catch (IOException ex) {
+            logger.error(Constants.IO_EXCEPTION_OCCURRED + ex.getMessage(), ex);
+            return null;

```

Comment:
Return null??

Reply:
null is been handled and an exception is been thrown